packages/core/src/connectors/walletConnect.ts

```diff
@@ -89,10 +89,20 @@ export class WalletConnectConnector extends Connector<
 
   async getProvider(create?: boolean) {
     if (!this.#provider || create) {
+      const rpc = !this.options.infuraId
+        ? this.chains.reduce(
+            (rpc, chain) => ({
+              ...rpc,
+              [chain.id]: chain.rpcUrls.default,
+            }),
+            {},
+          )
+        : undefined
+
       const WalletConnectProvider = (
         await import('@walletconnect/ethereum-provider')
       ).default
-      this.#provider = new WalletConnectProvider(this.options)
+      this.#provider = new WalletConnectProvider({ ...this.options, rpc })
```

Comment:
How would this work if someone set up their connector like this?

```ts
new WalletConnectConnector({
  options: {
    rpc: { 1: 'https://eth-mainnet.mycoolrpc.com/v2' }
  },
})
```

Seems like the defined `rpc` would get overridden because `infuraId` isn't defined.